src/fsverity/ioctl.rs

```diff
@@ -71,8 +75,33 @@ pub fn fs_ioc_measure_verity<F: AsFd, H: FsVerityHashValue>(fd: F) -> Result<H>
     }
 
     if digest.digest_algorithm != digest_algorithm || digest.digest_size != digest_size {
-        Err(std::io::Error::from(std::io::ErrorKind::InvalidData))?
+        Err(Errno::ILSEQ)
     } else {
         Ok(digest.digest)
     }
 }
+
+/// Like [`fs_ioc_measure_verity`] but does not fail if the file does not have fsverity enabled.
+pub fn fs_ioc_measure_verity_optional<F: AsFd, H: FsVerityHashValue>(fd: F) -> Result<Option<H>> {
+    match fs_ioc_measure_verity(fd) {
+        Ok(e) => Ok(Some(e)),
+        // See e.g. `lcfs_fd_measure_fsverity()`
+        Err(Errno::NODATA | Errno::NOSYS | Errno::NOTTY) => Ok(None),
```

Comment:
I really think `ENOSYS` is only gonna show up if `ioctl()` itself is not a supported system call (ie: never).  According to the docs:

> EOPNOTSUPP: the kernel was not configured with fs-verity support, or the filesystem superblock has not had the ‘verity’ feature enabled on it. (See [Filesystem support](https://docs.kernel.org/filesystems/fsverity.html#filesystem-support).)

Reply:
(If we change it here let's change the C composefs code too, I was just mirroring that)

I think you're probably right at a baseline; my suspicion is where some of this `ENOSYS` usage comes in is dealing with seccomp filters.

Dunno, how strongly do you feel? I am OK to drop it in both places, but it feels like a very-low-value, low-but-nonzero-risk case.

Or I guess we could just drop it here, and *not* change the composefs C code, and see if we ever hit it anywhere as this code gets used more widely...